dace/sdfg/sdfg.py

```diff
@@ -502,6 +502,8 @@ class SDFG(ControlFlowRegion):
                                            default=False,
                                            desc="Whether the SDFG contains explicit control flow constructs")
 
+    metadata = Property(dtype=dict, desc="Metada attached to the SDFG", default=None, allow_none=True)
```

Comment:
We already use an implicit property called metadata, afaik - during serialization, specifically. If that's the case, this may actually lead to conflicts possibly - can you double check this? (grep `metadata` on the repo and check conflicts)